microsoft/testsuites/core/azure_image_standard.py

```diff
@@ -1056,33 +1057,81 @@ def verify_serial_console_is_enabled(self, node: Node) -> None:
-        This test verifies that root bash history is either non-existent or empty in
-        the image.
+        This test verifies that bash/shell history files of all the bash/shell users
+        are either non-existent or empty in the image.
 
         Steps:
-        1. Check if /root/.bash_history file exists. If it doesn't exist, the test
-           passes as this indicates the image is properly prepared.
-        2. If /root/.bash_history exists, verify it is empty. If not empty, the test
+        1. Get all the bash/shell users' main directory from /etc/passwd file.
+        2. Using command 'find <user_main_dir> -type f
+           (-name ".*sh_history" -o -name ".history")' to get all the history files.
+        3. Using command 'ls -lt <history_file>' to check if the history file exists.
+        4. If it doesn't exist, the test passes as this indicates the image is properly
+           prepared.
+        5. If the history file exists, verify it is empty. If not empty, the test
            fails as bash history should be cleared.
+
+        The history file name of the users of "/bin/bash" and "/bin/sh" is
+        ".bash_history". The following shell types and their history file names
+        are listed below:
+        /bin/tcsh: .history
+        /bin/csh: .history
+        /bin/zsh: .zsh_history
+        /bin/ksh: .sh_history
+        /bin/dash: .sh_history
+        /bin/ash: .sh_history
+        /bin/pdksh: .sh_history
+        /bin/mksh: .sh_history
         """,
         priority=1,
         use_new_environment=True,
         requirement=simple_requirement(supported_platform_type=[AZURE, READY, HYPERV]),
     )
     def verify_bash_history_is_empty(self, node: Node) -> None:
-        path_bash_history = "/root/.bash_history"
-        cmd_result = node.execute(f"ls -lt {path_bash_history}", sudo=True, shell=True)
-        if 0 == cmd_result.exit_code:
-            cat = node.tools[Cat]
-            bash_history = cat.read(path_bash_history, sudo=True)
-            # "bash_history is not empty" is the failure triage pattern. Please be
-            # careful when changing this string.
-            assert_that(bash_history).described_as(
-                "/root/.bash_history is not empty. This could include private "
-                "information or plain-text credentials for other systems. It might be "
-                "vulnerable and exposing sensitive data. Please remove the bash "
-                "history completely using command 'sudo rm -f ~/.bash_history'."
-            ).is_empty()
+        remote_node = cast(RemoteNode, node)
+        current_user = str(remote_node.connection_info.get("username"))
+
+        # Get the bash and shell users' main directory from /etc/passwd file
+        cat = node.tools[Cat]
+        passwd_file = "/etc/passwd"
+        outputs = cat.read_with_filter(passwd_file, current_user, True, True, True)
+        for line in outputs.splitlines():
+            parts = line.strip().split(":")
```

Comment:
Private an example above, and use regexp.

Reply:
Done